### ISSUE TITLE
Use Nouns Instead of Verbs in Endpoints

### ISSUE THREAD

**Issue by pivee** [NaveenMare/modern-walk-expressjs-api#5]
```ts
@Route("api/v1/products")
@Tags("Products")
export default class ProductsController {
  ...

  @Post("/add")  ⬅️
  public addNewProduct(@Body() data: any) {
    return this.productsService.addNewProduct(data);
  }

  ...
}
```

When you're designing a REST API, you should not use verbs in the endpoint paths. The endpoints should use nouns, signifying what each of them does.

This is because HTTP methods such as GET, POST, PUT, PATCH, and DELETE are already in verb form for performing basic CRUD (Create, Read, Update, Delete) operations.

GET, POST, PUT, PATCH, and DELETE are the commonest HTTP verbs. There are also others such as COPY, PURGE, LINK, UNLINK, and so on.

So, for example, an endpoint should not look like this:

https://mysite.com/getPosts or https://mysite.com/createPost

Instead, it should be something like this: https://mysite.com/posts

In short, you should let the HTTP verbs handle what the endpoints do. So GET would retrieve data, POST will create data, PUT will update data, and DELETE will get rid of the data.

Source:
[REST API Best Practices – REST Endpoint Design Examples](https://www.freecodecamp.org/news/rest-api-best-practices-rest-endpoint-design-examples/)

**Comment by NaveenMarasinghe** [comment]
noted
